dist/ng-currency.js

```diff
@@ -57,6 +57,19 @@ angular.module('ng-currency', [])
                     }
                 }
 
+                function reformatViewValue(){
+                    var formatters = ngModel.$formatters,
+                        idx = formatters.length;
+
+                    var viewValue = ngModel.$modelValue;
+                    while (idx--) {
+                      viewValue = formatters[idx](viewValue);
+                    }
+
+                    ngModel.$setViewValue(viewValue);
```

Comment:
So the Angular runs the validators after running the formatters https://github.com/angular/angular.js/blob/master/src/ng/directive/ngModel.js#L828. So I would consider triggering a re-validation here as well because the `$viewValue` will have been changed by the formatters.

Maybe do `ngModel.$viewValue = viewValue;` here instead, and then below inside the `blur` call back, move the `ngModel.$commitViewValue();` to be after `reformatViewValue();`. The benefit of doing that is that `$commitViewValue` will run the validators.

What do you think?